### DISCUSSION TITLE
SH-RPi  manual installation in VenusOS on Rpi3B+

### DISCUSSION THREAD

**Discussion by virtuvas** [hatlabs/discussions#10] [General]
hello,

posted the same in a direct message chat we had a couple of months ago in signalK-dev but you probably missed that, np, just re-posting a light version of it here for your views.

So, got the SH-RPi plugged onto my Rpi 3B+, powered from N2K bus, all OK, pi works.
However CAN doesnt as I cannot get the overlays running as I cannot compile them in Venus OS (a v.light ver compared to rasbpian). further some paths are different, so auto script doesn't work anyway.
Wondering if you could post the two/three overlays so that I can manually get them in place and see if I can get can to work and then signalk to sent stuff to influx...

alternatively I have to download a raspbian OS ver, install it on my pi, compile/run the SH-RPi s/w and then manually copy the overlays over to the VenusOS install. bit messy, so wonder if you can help a bit there!

cheers

Vassilis

**Comment by mairas** [comment]
Hi! I'm so very sorry about the late reply - completely missed the message.

I've added the precompiled device tree overlays to the repo:

https://github.com/hatlabs/SH-RPi-daemon/tree/main/configs/binaries

**Comment by virtuvas** [comment]
Good morning, no worries Matti,

 

In the meantime, I installed a standalone rapbian on my rpi4, got them
compiled and reverted to my VenusOS. 

BTW, not as straight forward as I thought:

Renamed files as overlays (venusOS have  3 character extension), edited
accordingly the config.txt and whatever else needed, no luck.

 

Run out of time, currently on non-boaty vacations. Will be back to base at
the end of August. 

Will discuss it with the VenusOS maintainers and will come back.

Hope I'll be able to get it running and then we can have instructions for a
Victron VenusOS installation.

Expect another email from me after the 25th.

 

Cheers

 

V.

**Comment by johan-jernstrom** [comment]
Hi @virtuvas !
Been a while :) Just wondering if you got it working in the end?
Goin to attempt it myself now but could definitely use some pointers! :)

**Reply by johan-jernstrom** [reply]
Also, @mairas if you could compile what you know around getting your HW to work in a victron setup that would be great! 
For example, would the precompiled overlays work on my Rpi 3B+ (says in the readme they are mode for Rpi4...)?

**Comment by virtuvas** [comment]
hi Johan,

actually haven't as up till now I didn't need a backup system (rpi 3B+ and waveshare hat works fine onboard for the last couple of years)
However just got a new 7in display and put it back together my backup rpi planning to start working on it in March when I have free time.
I think these libs compiled by Matti  wont work in VenusOS fullstop. 
I have a rpi4 2GB and a rpi3B+ and I'm pretty sure I used the 4 for the tests.

would be keen to know if you get any leads yourself!

cheers
V

**Reply by johan-jernstrom** [reply]
@virtuvas great to hear!
I started working on it yesterday and actually has it runing now. 
At first, I had struggle getting i2c to work and downloaded the overlay for it from here: https://github.com/raspberrypi/firmware/blob/master/boot/overlays/i2c1.dtbo)
However, it wasnt until I followed the steps in this post: https://community.victronenergy.com/idea/41860/getting-i2c-running-for-humidity-and-temperature-s.html I got it working, not sure I needed the extra overlay or not.
I also had to install a bunch of extra python modules using python pip, and slightly modify the code.

The only thing I have got left is to run it as a service so the daemon starts on boot. If I have the time, I might also make it an installable mod using kwindrems setuphelper (maybe as a fork of this: https://github.com/kwindrem/ShutdownMonitor)

I also did not activate the RTC...
